app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -86,8 +86,14 @@ EOF
 elif [ -f /etc/debian_version ]; then
   <%= save_to_file('/etc/apt/sources.list.d/foreman_registration.list', @repo) %>
 <% if @repo_gpg_key_url.present? -%>
-  apt-get -y install ca-certificates gpg
-  curl --silent --show-error --output /etc/apt/trusted.gpg.d/client.asc <%= shell_escape @repo_gpg_key_url %>
+<# "apt 1.2.35" on Ubuntu 16.04 does not support storing GPG public keys in "/etc/apt/trusted.gpg.d/" in ASCII format #>
+  if [ $(. /etc/os-release ; echo "$VERSION_ID") = "16.04" ]; then
```

Comment:
You want to be safe and deal with the situation where it's empty (instead of the script crashing):
```suggestion
  if [ "$(. /etc/os-release ; echo "$VERSION_ID")" = "16.04" ]; then
```

Reply:
Applied.